### ISSUE TITLE
Release images to FluxChartRelease resources

### ISSUE THREAD

**Issue by squaremo** [weaveworks/flux#985]
`FluxChartRelease` resources are effectively workload controllers, or will usually be, since they will typically

 - refer to a chart that includes at least one core workload controller
 - specify particular images and tags (versions) as `values`

To support release automation, the flux machinery will need to understand how to interpret these resources, and how to update them.

**Comment by tamarakaufler** [comment]
## Investigation

Some possible formats of image info in Helm Chart's values.yaml:

```
image: memcached:1.5.6-alpine
```
```
image: itzg/minecraft-server
imageTag: latest
```
```
image:
  repository: quay.io/monax/db
  tag: 0.17.1
```
```
image:
  registry: docker.io
  repository: bitnami/mariadb
  tag: 10.1.32
```
```
artifactory:
  image:
    repository: "docker.bintray.io/jfrog/artifactory-pro"
    version: 5.9.1
```
```
jupyter:
  image:
    repository: "daskdev/dask-notebook"
    tag: "0.17.1"
```

**Comment by tamarakaufler** [comment]
There is a lot of variation in how the image information can be provided in a Chart. We can provide solution that will cope with most of the variations. Alternatively we can select one format meaning users would need change their Charts to use this particular format of the image info.

A) The former will result in:

> higher complexity
>        vs
> allowing flexible image info format with generally no need to alter the Chart
> 

B) The latter option will result in:

> simpler implementation
> vs
> possible need to alter a Chart to cope with our chosen format(s)

I think the A) option is preferable, especially as it is possible to adjust the level of complexity based on which of the image info formats should be supported.  

#### Other than FluxHelmRelease custom resources

To extend image automation to other CRDs (eg https://github.com/openfaas-incubator/faas-o6s/blob/master/artifacts/figlet-armhf.yaml), we should also consider having the image info directly under the Spec, ie not under Spec.Values.

**Comment by tamarakaufler** [comment]
### Implementation thoughts:

https://docs.google.com/document/d/1AoO8nMY_Ju9gh2LTgKgUbHFTEmV3ighDO4QKG2m9eTk/edit#heading=h.k9ts9ic80ezl

**Comment by tamarakaufler** [comment]
The selected approach is to register a new resource kind and provide the implementation specifics for a kind during the registration.
github branch: https://github.com/weaveworks/flux/tree/985-release-images-to-FHRs-v2

 
I encountered a problem with circular import, that I was not able to solve quickly, so need to rethink the design. In the meantime, as per agreement in WRCN weekly meetings (2018-5-01), I shall provide an implementation only for FHR Custom Resources containing image information in the format Spec.Values.image (eg quai.io/weaveworks/server-ui:master:a000001):
github branch: https://github.com/weaveworks/flux/tree/985-release-images-for-server-ui-FHR

**Comment by tamarakaufler** [comment]
Slow progress due to some issues in the current fluxd implementation, like duplication and extra complexity.

**Comment by squaremo** [comment]
> duplication and extra complexity

#1074 will help with that.

**Comment by squaremo** [comment]
Implemented in #1111.

**Comment by squaremo** [comment]
(should probably wait until that's merged)
